include/mbgl/map/map.hpp

```diff
@@ -159,6 +159,7 @@ class Map : private util::noncopyable {
     // Layers
     style::Layer* getLayer(const std::string& layerID);
     void addLayer(std::unique_ptr<style::Layer>, const optional<std::string>& beforeLayerID = {});
```

Comment:
remove the optional `beforeLayerID` and combine beforeOrAfter in the insert method?